src/client/Dockerfile

```diff
@@ -1,8 +1,17 @@
 FROM node:11.10-alpine as builder
 WORKDIR '/app'
+
+ARG CI
+ARG REACT_APP_BUILD_VERSION
+
 COPY ./package*.json ./
 RUN npm install
 COPY . .
+
+ENV CI=$CI
+ENV REACT_APP_BUILD_VERSION=$REACT_APP_BUILD_VERSION
+
+RUN npm run test -- --coverage
```

Comment:
Client unit tests run within the container build stage. Its one of those _better practices_

Reply:
Very nice. If the build fails does it stop the process and put the build report somewhere?

Reply:
If the build fails, it will be within the `Build images` stage log. It will be the last item in the logs showing the failure. 
As the steps are all within the `DockerFile`, it should be trivial to reproduce by running the `docker-compose build client` step. I like things being in reproducible docker steps.